nixos/modules/services/admin/nomachine.nix

```diff
@@ -0,0 +1,258 @@
+    # nxexec (symlink is not a folder)
+    cp -r ${nxserver}/NX/scripts.static/* /var/lib/nxserver/scripts/
+    for i in /var/lib/nxserver/scripts/restricted/*.sh; do
+      file=$(basename "$i")
+      if [ "$file" != "nxfunct.sh" ] && [ "$file" != "nxlogrotate.sh" ]; then
+        chown root:root "$i"
+        chmod u+s "$i"
+      fi
+    done
+
+    # nxserver can only run after the previous commands
+
+    if [ ! -f /etc/NX/uuid ]; then
+      ${nxserver}/NX/bin/nxkeygen -u > /var/lib/nxserver/etc/uuid
+    fi
+
+    # After this if block the node license is not yet valid
+    # This is done in the next license block
+    if [ ! -f /var/lib/nxserver/etc/node.lic ]; then
+      cp ${nxserver}/NX/etc.static/node.lic.sample /var/lib/nxserver/etc
+      /etc/NX/nxnode --validate
+      mv /var/lib/nxserver/etc/node.lic.sample /var/lib/nxserver/etc/node.lic
+      chown nx:root /var/lib/nxserver/etc/node.lic
+      chmod 0400 /var/lib/nxserver/etc/node.lic
+    fi
+
+    if [ ! -f /var/lib/nxserver/etc/server.lic ]; then
+      cp ${nxserver}/NX/etc.static/server.lic.sample /var/lib/nxserver/etc
+      /etc/NX/nxserver --validate
+      mv /var/lib/nxserver/etc/server.lic.sample /var/lib/nxserver/etc/server.lic
+      chown nx:root /var/lib/nxserver/etc/server.lic
+      chmod 0400 /var/lib/nxserver/etc/server.lic
+
+      # This fixed the node license
+      /etc/NX/nxserver --validatenode
+    fi
+
+    # This inits the database with the localhost_4000 display entry
+    /etc/NX/nxserver --addtoredis
+
+    if [ ! -f /var/lib/nxserver/etc/NX/keys/host/nx_host_rsa_key ]; then
+      ${nxserver}/NX/bin/nxkeygen -k /var/lib/nxserver/etc/keys/host/nx_host_rsa_key -c /var/lib/nxserver/etc/keys/host/nx_host_rsa_key.crt
+      chown -R nx:root /var/lib/nxserver/etc/keys
+      chmod 0400 /var/lib/nxserver/etc/keys/host/nx_host_rsa_key
+    fi
+  '';
+
+  serverConf = ''
+    ConfigFileVersion 4.0
+
+    EnableDebug ${toString cfg.enableDebug}
+    SessionLogLevel ${toString cfg.debugLevel}
+
+    AvailableSessionTypes unix-remote,unix-console,unix-default,unix-application,physical-desktop,shadow
+
+    EnablePasswordDB 0
+
+    Section "Server"
+
+    Name "Connection to localhost"
+    Host 127.0.0.1
+    Protocol NX
+    Port 4000
+    Authentication password
+
+    EndSection
+  '';
+
+  nodeConf = ''
+    ConfigFileVersion 4.0
+
+    EnableDebug ${toString cfg.enableDebug}
+    SessionLogLevel ${toString cfg.debugLevel}
+
+    AvailableSessionTypes unix-remote,unix-console,unix-default,unix-application,physical-desktop,shadow
+
+    EnableSmartcardSharing 1
+    EnableCUPSSupport 0
+    CUPSBinPath ""
+    CUPSSbinPath ""
+    CUPSBackendPath ""
+
+    EnableEGLCapture 0
+
+    #CommandFuser ${pkgs.psmisc}/bin/fuser
+    #CommandXdpyInfo ${pkgs.xorg.xdpyinfo}/bin/xdpyinfo
+    #CommandXmodmap ${pkgs.xorg.xmodmap}/bin/xmodmap
+    #CommandXkbComp ${pkgs.xorg.xkbcomp}/bin/xkbcomp
+    #CommandXauth ${nxserver}/NX/bin/nxauth
```

Comment:
I'm guessing these are comments, so maybe we should get rid of or uncomment them?